operator/Makefile

```diff
@@ -40,35 +73,49 @@ IMAGE_VERSION="${IMAGE_VERSION}" \
 NAMESPACE="${NAMESPACE}" \
 deploy/deploy-kiali-operator.sh
 
-# Remove the Kiali operator resources from the cluster along with Kiali itself
-operator-remove: purge-kiali
+## operator-delete: Remove the Kiali operator resources from the cluster along with Kiali itself
+operator-delete: purge-kiali
 	@echo Remove Operator
-	oc delete --ignore-not-found=true all,sa,deployments,clusterroles,clusterrolebindings,customresourcedefinitions --selector=app=kiali-operator -n ${OPERATOR_NAMESPACE}
-	oc delete --ignore-not-found=true namespace ${OPERATOR_NAMESPACE}
+	${OC} delete --ignore-not-found=true all,sa,deployments,clusterroles,clusterrolebindings,customresourcedefinitions --selector=app=kiali-operator -n ${OPERATOR_NAMESPACE}
+	${OC} delete --ignore-not-found=true namespace ${OPERATOR_NAMESPACE}
 
-# Create a Kiali CR to the cluster, informing the Kiali operator to install it.
-deploy-kiali:
-	@echo Deploy Kiali
-	oc apply -n ${OPERATOR_NAMESPACE} -f deploy/kiali/kiali_cr_dev.yaml
+## secret-create: Create a Kiali secret using CREDENTIALS_USERNAME and CREDENTIALS_PASSPHRASE.
+secret-create:
+	@echo Create the secret
+	${OC} create secret generic kiali -n ${NAMESPACE} --from-literal "username=${CREDENTIALS_USERNAME}" --from-literal "passphrase=${CREDENTIALS_PASSPHRASE}"
+	${OC} label secret kiali app=kiali -n ${NAMESPACE}
+
+## secret-delete: Delete the Kiali secret.
+secret-delete:
+	@echo Delete the secret
+	${OC} delete --ignore-not-found=true secret --selector=app=kiali -n ${NAMESPACE}
+
+## kiali-create: Create a Kiali CR to the cluster, informing the Kiali operator to install Kiali.
+kiali-create: secret-create
```

Comment:
This always calls _secret-create_ which is not needed for AUTH_STRATEGY=openshift. As mentioned in deploy-kiali-operator.sh

Reply:
Correct. I was assuming if you create the secret with the makefile you will also delete it (using secret-delete or kiali-delete). But I could add some conditional statement here that the secret will only be created for you if you set AUTH_STRATEGY to login.

Reply:
This is fixed - if AUTH_STRATEGY is login, that target will also invoke secret-create; otherwise it will not. The delete target still always deletes the secret to avoid forcing the user from having to set auth strategy when calling kiali-delete.